### PR TITLE
CLN: Fix exception causes in datetimelike.py

### PR DESCRIPTION
**Pull request by cool-RR**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/datetimelike.py
Comment (jbrockmendel) [inline pandas/core/arrays/datetimelike.py]: nitpick: can you make this "err" instead of "e"; i really dont like 1-letter names

Comment (gfyoung) [reply pandas/core/arrays/datetimelike.py]: Eh...I would leave it alone for this PR.  I understand having an aesthetic preference, but `e` is strewn all over the place in this file, and I don't see why we should nitpick this one instance and hold this PR up because of that.

If we have a generally accepted (for this repository) a stylistic preference for variable names for exceptions, then fine, but I'm not aware of that at this point.

**Comment by cool-RR** [comment]
Thanks for reviewing guys. I'm ambivalent towards the `err` change, so if you'd like me to change, I'll change it. If there's anything else I need to do about this PR, let me know.

**Comment by gfyoung** [comment]
Thanks @cool-RR, and welcome to pandas! 🎉

`err` naming question notwithstanding, it looks good!

File: pandas/core/arrays/datetimelike.py
Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py]: yah, not a deal-breaker.  i think of this as a "if you have to push another commit anyway, this would be nice"

**Comment by jbrockmendel** [comment]
@cool-RR is there something here we should be testing?

**Comment by cool-RR** [comment]
If you mean testing that the `__cause__` attribute is set correctly, I think that's overkill. The only kind of testing that's useful is having test code that causes the exception, reaches the `raise` and ensures the correct exception gets raised, but that's orthogonal to this PR, i.e. it's needed about as much as it's needed without this PR.

**Comment by jbrockmendel** [comment]
sounds good, LGTM

**Comment by cool-RR** [comment]
Cool. Any next steps needed before merging?

**Review by gfyoung — APPROVED** [review]

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
I would be on board with doing this more generally if there's a way to identify easily

**Comment by WillAyd** [comment]
Thanks @cool-RR

**Pull request merged**
